File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        res = 1
        rems = set()
        lastRem = 1 % k
        rems.add(lastRem)
        while lastRem != 0:
            res += 1
            curRem = (((lastRem * 10) % k) + 1) % k
            if curRem in rems:
                res = -1
                break
            rems.add(curRem)
            lastRem = curRem
        return res

File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        int res = 1;
        unordered_set<int> rems;
        int lastRem = 1 % k;
        rems.insert(lastRem);
        while(lastRem != 0)
        {
            res++;
            int curRem = (((lastRem * 10) % k) + 1) % k;
            if(rems.find(curRem) != rems.end())
            {
                res = -1;
                break;
            }
            rems.insert(curRem);
            lastRem = curRem;
        }
        return res;
    }
};